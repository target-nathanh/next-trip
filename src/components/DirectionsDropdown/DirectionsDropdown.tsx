import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import find from 'lodash/find';
import { Direction } from '../../types';
import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles({
  select: {
    width: '33%',
    fontSize: '20px',
  },
});

interface DirectionDropdownProps {
  directions: Direction[];
  selectedDirection?: Direction;
  onSelectDirection: (direction: Direction) => void;
}

const DirectionsDropdown: React.FC<DirectionDropdownProps> = ({
  selectedDirection,
  directions,
  onSelectDirection,
}) => {
  const classes = useStyles();
  const onChange = (event: React.ChangeEvent<{ value: unknown }>) => {
    const selectedDirection = find(directions, { direction_id: event.target.value }) as Direction;
    if (selectedDirection) {
      onSelectDirection(selectedDirection);
    }
  };
  return (
    <Select className={classes.select} value={selectedDirection?.direction_id} onChange={onChange}>
      {directions?.map((direction: Direction) => (
        <MenuItem key={direction.direction_id} value={direction.direction_id}>
          {direction.direction_name}
        </MenuItem>
      ))}
    </Select>
  );
};

export default DirectionsDropdown;
