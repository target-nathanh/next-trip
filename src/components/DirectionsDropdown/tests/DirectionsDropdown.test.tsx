import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import directions from '../../../mockData/directionsMock.json';
import DirectionsDropdown from '../DirectionsDropdown';
import { MemoryRouter } from 'react-router-dom';

describe('DirectionsDropdown', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <DirectionsDropdown
          directions={directions}
          selectedDirectionId={directions[0].direction_id}
          onSelectDirection={() => {}}
        />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls onSelectDirection when new direction is selected', async () => {
    const onSelectDirectionMock = jest.fn();
    const { getByRole } = render(
      <MemoryRouter>
        <DirectionsDropdown
          directions={directions}
          selectedDirectionId={directions[0].direction_id}
          onSelectDirection={onSelectDirectionMock}
        />
      </MemoryRouter>
    );
    fireEvent.mouseDown(getByRole('button'));

    const option = await screen.findByText(directions[1].direction_name);
    fireEvent.click(option);
    expect(onSelectDirectionMock).toBeCalledWith(directions[1].direction_id);
  });
});
