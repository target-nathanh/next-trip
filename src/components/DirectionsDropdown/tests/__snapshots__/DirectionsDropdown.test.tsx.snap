// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionsDropdown matches snapshot 1`] = `
<div
  class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-1"
>
  <div
    aria-haspopup="listbox"
    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    role="button"
    tabindex="0"
  >
    Northbound
  </div>
  <input
    aria-hidden="true"
    class="MuiSelect-nativeInput"
    placeholder="Select direction"
    tabindex="-1"
    value="0"
  />
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
